include/zephyr/drivers/pinctrl/pinctrl_nrf_common.h

```diff
@@ -0,0 +1,259 @@
+	 ((NRF_LP_ENABLE * DT_PROP(node_id, low_power_enable)) << NRF_LP_POS) |\
+	 (DT_PROP(node_id, nordic_invert) << NRF_INVERT_POS) |		       \
+	 Z_GET_CLOCKPIN_ENABLE(node_id, prop, idx, p_node_id) |		       \
+	 Z_GET_GPD_FAST_ACTIVE1(p_node_id)				       \
+	),
+
+/**
+ * @brief Utility macro to initialize state pins contained in a given property.
+ *
+ * @param node_id Node identifier.
+ * @param prop Property name describing state pins.
+ */
+#define Z_PINCTRL_STATE_PINS_INIT(node_id, prop)			       \
+	{DT_FOREACH_CHILD_VARGS(DT_PHANDLE(node_id, prop),		       \
+				DT_FOREACH_PROP_ELEM_VARGS, psels,	       \
+				Z_PINCTRL_STATE_PIN_INIT, node_id)}
+
+/**
+ * @brief Utility macro to obtain pin function.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_FUN(pincfg) (((pincfg) >> NRF_FUN_POS) & NRF_FUN_MSK)
+
+/**
+ * @brief Utility macro to obtain pin clockpin enable flag.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_CLOCKPIN_ENABLE(pincfg) \
+	(((pincfg) >> NRF_CLOCKPIN_ENABLE_POS) & NRF_CLOCKPIN_ENABLE_MSK)
+
+/**
+ * @brief Utility macro to obtain GPD_FAST_ACTIVE1 flag
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_GPD_FAST_ACTIVE1(pincfg) \
+	(((pincfg) >> NRF_GPD_FAST_ACTIVE1_POS) & NRF_GPD_FAST_ACTIVE1_MSK)
+
+/**
+ * @brief Utility macro to obtain pin inversion flag.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_INVERT(pincfg) (((pincfg) >> NRF_INVERT_POS) & NRF_INVERT_MSK)
+
+/**
+ * @brief Utility macro to obtain pin low power flag.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_LP(pincfg) (((pincfg) >> NRF_LP_POS) & NRF_LP_MSK)
+
+/**
+ * @brief Utility macro to obtain pin drive mode.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_DRIVE(pincfg) (((pincfg) >> NRF_DRIVE_POS) & NRF_DRIVE_MSK)
+
+/**
+ * @brief Utility macro to obtain pin pull configuration.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_PULL(pincfg) (((pincfg) >> NRF_PULL_POS) & NRF_PULL_MSK)
+
+/**
+ * @brief Utility macro to obtain port and pin combination.
+ *
+ * @param pincfg Pin configuration bit field.
+ */
+#define NRF_GET_PIN(pincfg) (((pincfg) >> NRF_PIN_POS) & NRF_PIN_MSK)
+
+/** @endcond */
+
+/* SoC-specific hooks. */
+
+/**
+ * @brief Hook called at the beginning of @ref pinctrl_configure_pins
+ *
+ * @param pins List of pins to be configured.
+ * @param pin_cnt Number of pins.
+ * @param reg Device register (optional).
+ *
+ * @retval 0 If succeeded
+ * @retval -errno Negative errno for other failures.
+ */
+int nrf_pinctrl_hook_start(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
+			   uintptr_t reg);
+
+/**
+ * @brief Hook called before GPIO properties configuration is written to register.
+ *
+ * @param pin Pointer to a structure describing pin that is currently handled.
+ * @param cfg Pointer to a structure describing current pin's configuration.
```

Comment:
It's not the current configuration, but the configuration to be applied. And as I understand it, the hook is allowed to alter it, right?

Reply:
True, I'll rephrase to `currently evaluated pin's configuration that will be applied`.